### DISCUSSION TITLE
Controlling USB connected hardware with Termux:API

### DISCUSSION THREAD

**Discussion by LobsterMan123** [termux/termux-app#3076] [General]
I'd like to do something where a device is connected to my tablet via usb cable and then control said device using Termux:API on my tablet (like maybe blink an LED on a breadboard).  Is this possible using Termux:API?

Next, I'd like to know if the code on Termux's wiki page for Termux:API counts/can be considered as a simple device driver for the example given (URL: https://wiki.termux.com/wiki/Termux:API).  This particular example seems to acquire device info of the USB device, but I'm not sure if more of actual device control can be realized using additional C programming knowledge which is what my questions revolve around.

Thanks.

LM

**Comment by sylirre** [comment]
[termux-usb](https://wiki.termux.com/wiki/Termux-usb) with custom libusb-based program, especially if you want to blink led on something. The linked page has a very basic example of C program which will work with `termux-usb`.

**Comment by AniketMoh** [comment]
Did you find out any method to blink led through terminal? We have the same project where we aren't allowed to use any sort of external hardware
